src/meta/Macro.h

```diff
@@ -168,6 +169,30 @@ class Macro {
     return str;
   }
 
+  static inline std::vector<std::string> enumeratePathsWithCustomMacros(const std::string& input, const std::map<std::string, std::vector<std::string>>& macroValues, int macroIndex = 0) {
+    auto it = macroValues.begin();
+    std::advance(it, macroIndex);
+    if (it == macroValues.end()) {
+      return std::vector<std::string>({input});
+    }
+    std::vector<std::string> results, nextSteps;
+    const auto searchString = nebula::common::format("{{s}}", {{"s", it->first}});
+    if (input.find(searchString) != std::string::npos) {
+      for (const auto& replacementVal : it->second) {
+        nextSteps.emplace_back(nebula::common::format(input, {{it->first, replacementVal}}));
+      }
+    } else {
+      nextSteps.emplace_back(input);
```

Comment:
have a little bit of concern on both readability and performance - if the result of combinations is large, likely become noticeable slow? we can get a sense of it after we have a reasonable test with the simulated size of the input.

for readability, do you think it's an idea to have a common independent method to produce all combinations of each Marco's index and use that to produce a new map for `format` to take? might be much easier to read

pseudo code in my mind: 
```
    macroIndices = common::combinations();
    for(indice in macroIndices){
         params = map<string, string> from indice;
         result.emplace_back(common::format(path, params));
    }
    return result;
```